File: src/main/java/com/ecommerce/offer/domain/model/Offer.java
package com.ecommerce.offer.domain.model;

import com.ecommerce.product.domain.model.Product;
import com.ecommerce.shared.domain.Quantity;
import com.ecommerce.shared.domain.Version;
import com.ecommerce.user.domain.model.UserId;

public record Offer(OfferId id, UserId userId, OfferStatus status, OfferPrice price, Quantity quantity, Version version, Product product) {

    public boolean isActive() {
        return status == OfferStatus.ACTIVE;
    }
}

File: frontend/src/types/offer.ts
import { Product } from '@src/types/product';

export interface OffersState {
  data: Offer[];
  total: number;
}

export interface OfferDetailsState {
  offer: Offer;
}
export interface AddOfferRequest {
  userId: string;
  price: string;
  quantity: string;
  productId: string;
}

export interface SearchOffersRequest {
  userId?: string;
  productCategory?: string;
  page: number;
  pageSize: number;
}

export interface ModifyOfferRequest {
  offerId: string;
  userId: string;
  version: number;
  price?: string;
  quantity?: string;
  productId?: string;
}
export interface Offer {
  id: string;
  userId: string;
  status: string;
  price: string;
  quantity: string;
  version: number;
  product: Product;
}
